test/getters.test.js

```diff
@@ -0,0 +1,300 @@
+import { jDataView } from "../src/jdataview";
+import {
+	compareBytes,
+	compareWithNaN,
+	getPrefilledJDataView,
+	b,
+	chr,
+	bufferToHex,
+} from "./test-helpers";
+import { describe, test, assert } from "vitest";
+
+describe("Getters", function () {
+	// getter = value || {value, check?, view?, args?}
+	function testGetters(type, getters) {
+		const view = getPrefilledJDataView();
+		test(type, function () {
+			getters.forEach(function (getter) {
+				if (typeof getter !== "object") {
+					getter = { value: getter };
+				}
+
+				const args = getter.args || [];
+				const contextView = getter.view || view;
+				const check = getter.check || assert.equal;
+				const value = getter.value;
+				const offset = contextView.tell();
+				const realValue = contextView["get" + type].apply(
+					contextView,
+					args
+				);
+
+				check(
+					realValue,
+					value,
+					"get" +
+						type +
+						"(" +
+						args.join(", ") +
+						") == " +
+						realValue +
+						" != " +
+						value +
+						" at offset " +
+						offset +
+						(getter.view
+							? " in view [" +
+							  getter.view
+									.getBytes(undefined, 0, true)
+									.join(", ") +
+							  "]"
+							: "")
+				);
+			});
+		});
+	}
+
+	testGetters("Bytes", [
+		{ args: [2, 1], value: [0xfe, 0xfd], check: compareBytes },
+		{
+			args: [3, 0, false],
+			value: [0xfd, 0xfe, 0xff],
+			check: compareBytes,
+		},
+	]);
+
+	testGetters("Char", [
+		chr(0xff),
+		chr(0xfe),
+		chr(0xfd),
+		chr(0xfc),
+		chr(0xfa),
+		chr(0),
+		chr(0xba),
+		chr(1),
+	]);
+
+	testGetters("String", [
+		{ args: [1, 0], value: chr(0xff) },
+		{ args: [1, 5], value: chr(0) },
+		{ args: [1, 7], value: chr(1) },
+		{
+			view: b(127, 0, 1, 65, 66),
+			args: [5],
+			value: chr(127) + chr(0) + chr(1) + chr(65) + chr(66),
+		},
+		{
+			view: b(0xd1, 0x84, 0xd1, 0x8b, 0xd0, 0xb2),
+			args: [undefined, undefined, "utf8"],
+			value: chr(1092) + chr(1099) + chr(1074),
+		},
+	]);
+
+	test(
+		"Big String",
+		function () {
+			const view = new jDataView(2000000);
+			assert.doesNotThrow(function () {
+				view.getString();
+			});
+		},
+		{ timeout: 5000 }

```

Comment:
Why does this need timeout? It would always throw synchronously if at all.

Reply:
I have a vague recollection form when I was converting the tests that it had a timeout for some reason.

My best guess is there might have been some old engine bug that caused the thread to hang on that call sometimes?

Anyway, seems like it's not useful now.